addons/hr_timesheet/wizard/hr_employee_delete_wizard.py

```diff
@@ -0,0 +1,66 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import api, fields, models, _
+
+
+class HrEmployeDeleteWizard(models.TransientModel):
+    _name = 'hr.employee.delete.wizard'
+    _description = 'Employee Delete Wizard'
+
+    employee_ids = fields.Many2many('hr.employee', string='Employees')
+    has_active_employee = fields.Boolean(string='Has Active Employee', compute='_compute_has_active_employee')
+    has_timesheet = fields.Boolean(string='Has Timesheet', compute='_compute_has_timesheet', compute_sudo=True)
+
+    @api.depends('employee_ids')
+    def _compute_has_timesheet(self):
+        self = self.with_context(active_test=False)
+        timesheet_read_group = self.env['account.analytic.line']._read_group([
+            ('employee_id', 'in', self.employee_ids.ids)],
+            ['employee_id'], ['employee_id']
+        )
+        timesheet_employee_map = set(employee_info['employee_id'][0] for employee_info in timesheet_read_group)
+        for wizard in self:
+            wizard.has_timesheet = timesheet_employee_map & set(wizard.employee_ids.ids)
+
+    @api.depends('employee_ids')
+    def _compute_has_active_employee(self):
+        for wizard in self:
+            wizard.has_active_employee = wizard.with_context(active_test=True).employee_ids
+
+    def action_archive(self):
+        self.ensure_one()
+        if len(self.employee_ids) != 1:
+            return self.employee_ids.toggle_active()
+        return {
+            'name': _('Employee Termination'),
+            'type': 'ir.actions.act_window',
+            'res_model': 'hr.departure.wizard',
+            'views': [[False, 'form']],
+            'view_mode': 'form',
+            'target': 'new',
+            'context': {
+                'active_id': self.employee_ids.id,
+                'toggle_active': True,
+            }
+        }
+
+    def confirm_delete(self):
+        self.ensure_one()
+        self.with_context(active_test=False).employee_ids.unlink()
```

Comment:
we could do the following code if the `active_test` is added in the field definition. 🙂 
```suggestion
        self.employee_ids.unlink()
```